src/main/java/org/jboss/aerogear/adm/TokenService.java

```diff
@@ -47,44 +39,58 @@
      * @return a String containing your auth token
      * @throws Exception if retrieving the Auth token fails
      */
-    public String getAuthToken(String clientId, String clientSecret) throws Exception
-    {
+    public String getAuthToken(String clientId, String clientSecret) throws Exception {
         // Encode the body of your request, including your clientID and clientSecret values.
-        String body = "grant_type="    + URLEncoder.encode(CLIENT_CREDENTIALS, UTF_8) + "&" +
-                "scope="         + URLEncoder.encode(MESSAGING_PUSH, UTF_8)     + "&" +
-                "client_id="     + URLEncoder.encode(clientId, UTF_8)             + "&" +
-                "client_secret=" + URLEncoder.encode(clientSecret, UTF_8);
+        String body = buildBody(clientId, clientSecret);
 
         // Generate the HTTPS connection. You cannot make a connection over HTTP.
         final HttpsURLConnection con = post(body);
 
         // Convert the response into a String object.
-        final String responseContent = parseResponse(con.getInputStream());
+        final String responseContent = Utilities.parseResponse(con.getInputStream());
 
-        // Create a new JSONObject to hold the access token and extract
-        // the token from the response.
-        final JSONObject parsedObject = new org.json.JSONObject(responseContent);
-        final String accessToken = parsedObject.getString("access_token");
+        final String accessToken = Utilities.getStringFromJson(responseContent, "access_token");
         return accessToken;
     }
 
+
+    private String buildBody(String clientId, String clientSecret) throws UnsupportedEncodingException {
+        StringBuilder builder = new StringBuilder();
+        builder
+                .append("grant_type=")
+                .append(URLEncoder.encode(Utilities.CLIENT_CREDENTIALS, Utilities.UTF_8) )
+                .append("&")
+                .append("scope=")
+                .append( URLEncoder.encode(Utilities.MESSAGING_PUSH, Utilities.UTF_8) )
+                .append("&")
+                .append("client_id=")
+                .append("URLEncoder.encode(clientId, Utilities.UTF_8)")
```

Comment:
@sebastienblanc is this string `"URLEncoder.encode(clientId, Utilities.UTF_8)"` really wanted ? :-)